src/components/TodoInfo/TodoInfo.tsx

```diff
@@ -0,0 +1,96 @@
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  togleStatus: (
+    id: number,
+    completed: boolean) => void;
+  deleteTodo: (id: number) => void;
+  changeInputText: (id: number, query: string) => void;
+};
+
+export const TodoInfo: React.FC<Props> = ({
+  todo,
+  togleStatus,
+  deleteTodo,
+  changeInputText,
+}) => {
+  const { title, completed, id } = todo;
+  const [isEditing, setIsEditing] = useState(false);
+  const [query, setQuery] = useState(title);
+
+  const handleInputChange = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter') {
+      setIsEditing(false);
+      setQuery(event.target.value);
+      if (!query) {
+        deleteTodo(id);
+
+        return;
+      }
+
+      changeInputText(id, query);
+    }
+
+    if (event.key === 'Escape') {
```

Comment:
Would be better to use switch...case for that.
Also, try not to hardcore the values, but create Enum for them